src/main/java/org/jfrog/hudson/pipeline/common/types/deployers/MavenDeployer.java

```diff
@@ -121,32 +120,34 @@ private static MavenDeployer createEmptyDeployer() {
      * Adds artifacts from the provided modules to the Deployed Maven Artifacts Summary Action.
      */
     public static void addDeployedArtifactsActionFromModules(Run build, String artifactoryUrl, List<Module> modules) {
+        List<DeployedMavenArtifact> curArtifacts = Lists.newArrayList();
         for (Module module : modules) {
-            if (module.getArtifacts() != null) {
-                addDeployedArtifactsAction(build, artifactoryUrl, module.getArtifacts());
+            if (module.getArtifacts() == null) {
+                continue;
+            }
+            for (Artifact artifact : module.getArtifacts()) {
+                curArtifacts.add(new DeployedMavenArtifact(artifactoryUrl, module.getRepository(),
+                        artifact.getRemotePath(), artifact.getName()));
             }
         }
+        addDeployedArtifactsToAction(build, curArtifacts);
     }
 
     /**
      * Adds artifacts from the provided DeployDetails map to the Deployed Maven Artifacts Summary Action.
      */
     public static void addDeployedArtifactsActionFromDetails(Run build, String artifactoryUrl, Map<String, Set<DeployDetails>> deployableArtifactsByModule) {
         deployableArtifactsByModule.forEach((module, detailsSet) -> {
+            // Add only if whole module contains maven artifacts.
             boolean isMaven = true;
-            List<Artifact> curArtifacts = Lists.newArrayList();
+            List<DeployedMavenArtifact> curArtifacts = Lists.newArrayList();
             for (DeployDetails curDetails : detailsSet) {
                 isMaven = (curDetails.getPackageType() == DeployDetails.PackageType.MAVEN) && isMaven;
```

Comment:
Wouldn't it be better to just return here, if the package type is not maven?
This will also make the ```if (isMaven) {``` at the bottom of the method redundant.